src/macaron/dependency_analyzer/cyclonedx.py

```diff
@@ -35,155 +46,608 @@ def deserialize_bom_json(file_path: Path) -> dict:
+
+        Yields
+        ------
+        CDXComponent
+            The dependencies as CycloneDX components.
+        """
+        try:
+            root_bom = deserialize_bom_json(root_bom_path)
+        except CycloneDXParserError as error:
+            logger.error(error)
+            return
 
-    Parameters
-    ----------
-    components : list[dict]
-        The dependency components.
-    root_component: dict | None
-        The root CycloneDX component.
+        if root_bom.components is None:
+            logger.error("The BOM file at %s misses components.", str(root_bom_path))
+            return
 
-    Returns
-    -------
-    dict
-        A dictionary where dependency artifacts are grouped based on "groupId:artifactId".
-    """
-    all_versions: dict[str, list[DependencyInfo]] = {}  # Stores all the versions of dependencies for debugging.
-    latest_deps: dict[str, DependencyInfo] = {}  # Stores the latest version of dependencies.
-    url_to_artifact: dict[str, set] = {}  # Used to detect artifacts that have similar repos.
-    for component in components:
+        dependencies: list[CDXDependency] = []
+
+        # Find dependencies in the root BOM file.
         try:
+            target_cdx_component = self.get_target_cdx_component(root_bom=root_bom, target_component=target_component)
+            for node in root_bom.dependencies:
+                if not isinstance(node, CDXDependency):
+                    continue
+                if recursive or (
+                    target_cdx_component and target_cdx_component.bom_ref and node.ref == target_cdx_component.bom_ref
+                ):
+                    if dep_on := node.dependencies:
+                        dependencies.extend(dep_on)
+        except AttributeError as error:
+            logger.debug(error)
+
+        # Find dependencies in child BOMs if they exist. Multi-module Java projects need this resolution.
+        child_bom_objects: list[Bom] = []
+        modules: set[str] = set()  # Stores all module dependencies.
+
+        for child_path in child_bom_paths or []:
+            try:
+                child_bom_objects.append(deserialize_bom_json(child_path))
+            except CycloneDXParserError as error:
+                logger.error(error)
+                continue
+
+        for bom in child_bom_objects:
+            if not bom.metadata or not bom.metadata.component:
+                continue
+            try:
+                target_bom_ref = bom.metadata.component.bom_ref
+                if target_bom_ref and target_bom_ref.value:
+                    modules.add(target_bom_ref.value)
+                for node in bom.dependencies:
+                    if not isinstance(node, CDXDependency):
+                        continue
+                    if node.ref == target_bom_ref or recursive:
+                        dependencies.extend(node.dependencies)
+                        break
```

Comment:
I wonder why for this part, `recursive` is put behind the OR while in this part, it's the opposite :thinking: 
https://github.com/oracle/macaron/blob/4e2c3ab8bceef002c534691910218773cf665a29/src/macaron/dependency_analyzer/cyclonedx.py#L575-L579